### PR TITLE
Create enroll lesson unit test

### PR DESCRIPTION
**Pull request by hasancan-code**
As mentioned in the [issue 250](https://github.com/bounswe/bounswe2022group2/issues/250) I will be implementing the unit tests for POST endpoint of enrolling lessons.

1. Successful enrolling to a lesson
2. Request with missing user_id
3. Request with missing lesson_id
4. Request with invalid user_id
5. Request with invalid lesson_id
6. Request with user_id which does not exist
7. Request with lesson_id which does not exist
8. Request with user_id which is already taken this course

### CODE REVIEW

File: practice-app/server/test/enroll_lesson.test.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+          expect(res.statusCode).toBe(200);
+        })
+        .end(done);
+    });
+    
+    it('should return the user with the given user_id with remaining enrolledLessons list and removed lesson with the given lesson_id', (done) => {
+  
+      request(app)
+        .post(enrollUrl)
+        .send({ user_id: user_ids[1], lesson_id: lesson_ids[3] })
```
Comment (egemenatikk) [inline practice-app/server/test/enroll_lesson.test.js:180]: I am not sure but I think there is no lesson with the id lesson_ids[3], so last test would give an error with status code 400.

Comment (hasancan-code) [reply practice-app/server/test/enroll_lesson.test.js:186]: I think this resolved now.

**Comment by egemenatikk** [comment]
Tests for enrolling a lesson endpoint look very comprehensive and nice. Thank you for your work @hasancan-code. 

I am not sure but I guess there is just one small mistake, I talked about it in the previous comment.

File: practice-app/server/test/enroll_lesson.test.js (hunk 1 above)
Comment (egemenatikk) [reply practice-app/server/test/enroll_lesson.test.js:186]: Yes, thank you.

**Review by egemenatikk — APPROVED** [review]
I cannot think anything else to add, tests look fine and error-free, I think it is ready to be merged.

**Pull request merged**